### PR TITLE
Multiple controllers support

### LINKED ISSUE #931: Feature: Support multiple joysticks / game controllers
DevilutionX currently only opens one joystick / game controller.
If multiple controllers are plugged in, it should process all of them.

Use case: A handheld retro devices with an external controller and HDMI out plugged in.

### PR DESCRIPTION
**Pull request by glebm**
Handle events from all connected controllers.

A minor issue: START+SELECT combo doesn't work at all on the external controller in batocera.linux.

Fixes #931

### CODE REVIEW

File: SourceX/controls/devices/game_controller.cpp [hunk 1]
```diff
@@ -1,36 +1,37 @@
...
-ControllerButton GameControllerToControllerButton(const SDL_Event &event)
+ControllerButton GameController::ToControllerButton(const SDL_Event &event)
 {
 	switch (event.type) {
 	case SDL_CONTROLLERAXISMOTION:
 		switch (event.caxis.axis) {
 		case SDL_CONTROLLER_AXIS_TRIGGERLEFT:
 			if (event.caxis.value < 8192) { // 25% pressed
-				sgbTriggerLeftDown = false;
+				trigger_left_is_down_ = false;
```
Comment (AJenbo) [inline SourceX/controls/devices/game_controller.cpp:22]: What is the odd name styling supposed to indicate?

File: SourceX/controls/controller.cpp [hunk 2]
```diff
@@ -28,34 +28,55 @@ ControllerButtonEvent ToControllerButtonEvent(const SDL_Event &event)
...
 {
-	bool result = false;
 #ifndef USE_SDL1
-	result = result || IsGameControllerButtonPressed(button);
+	if (GameController::IsPressedOnAnyController(button))
+		return true;
 #endif
 #if HAS_KBCTRL == 1
-	result = result || IsKbCtrlButtonPressed(button);
+	if (IsKbCtrlButtonPressed(button)) return true;
```
Comment (qndel) [inline SourceX/controls/controller.cpp:53]: add new line

**Review by qndel — CHANGES_REQUESTED** [review]

File: SourceX/controls/devices/game_controller.cpp (hunk 1 above)
Comment (glebm) [reply SourceX/controls/devices/game_controller.cpp:22]: Private member variable

This is Google style.

What would you like me to rename them?

File: SourceX/display.cpp [hunk 3]
```diff
@@ -121,8 +123,14 @@ bool SpawnWindow(const char *lpWindowName)
...
 	SDL_EnableUNICODE(1);
 #endif
-#if defined(USE_SDL1) || defined(__SWITCH__)
-	InitController();
+#ifdef USE_SDL1
+	// On SDL 1, there are no ADDED/REMOVED events.
+	// Always try to initialize the first joystick.
+	Joystick::Add(0);
+#ifdef __SWITCH__
+	// TODO: Is this really needed on the Switch?
```
Comment (AJenbo) [inline SourceX/display.cpp:131]: I was at the time, but I don't know if the SDL port has fixed it by now.

File: SourceX/controls/controller.cpp (hunk 2 above)
Comment (AJenbo) [reply SourceX/controls/controller.cpp:53]:
```suggestion
	if (IsKbCtrlButtonPressed(button))
		return true;
```

File: SourceX/display.cpp (hunk 3 above)
Comment (glebm) [reply SourceX/display.cpp:131]: Yeah, this is something to investigate for someone with a hacked Switch
This PR preserves the previous behaviour

File: SourceX/controls/controller.cpp (hunk 2 above)
Comment (glebm) [reply SourceX/controls/controller.cpp]: Done

File: SourceX/display.cpp (hunk 3 above)
Comment (AJenbo) [reply SourceX/display.cpp:131]:
```suggestion
	// TODO: There is a bug in SDL2 on Switch where it does not repport controllers on startup (Jan 1, 2020)
```

**Pull request merged**